src/ynEIGEN/AssetRegistry.sol

```diff
@@ -0,0 +1,323 @@
+        IynEigen ynEigen;
+        address admin;
+        address pauser;
+        address unpauser;
+    }
+
+    function initialize(Init calldata init)
+        public
+        notZeroAddress(address(init.rateProvider))
+        notZeroAddress(address(init.admin))
+        notZeroAddress(address(init.pauser))
+        notZeroAddress(address(init.unpauser))
+        initializer {
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, init.admin);
+        _grantRole(PAUSER_ROLE, init.pauser);
+        _grantRole(UNPAUSER_ROLE, init.unpauser);
+
+        for (uint256 i = 0; i < init.assets.length; i++) {
+            if (address(init.assets[i]) == address(0)) {
+                revert ZeroAddress();
+            }
+            assets.push(init.assets[i]);
+            _assetData[init.assets[i]] = AssetData({
+                active: true
+            });
+        }
+
+        rateProvider = init.rateProvider;
+        eigenStrategyManager = init.eigenStrategyManager;
+        ynEigen = init.ynEigen;
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  ASSET MANAGEMENT  --------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /**
+     * @notice Adds a new asset to the system.
+     * @dev Adds an asset to the _assetData mapping and sets it as active. This function can only be called by the strategy manager.
+     * @param asset The address of the ERC20 token to be added.
+     */
+    function addAsset(IERC20 asset) public onlyRole(ASSET_MANAGER_ROLE) whenNotPaused {
+        require(!_assetData[asset].active, "Asset already active");
+
+        assets.push(asset);
+
+        _assetData[asset] = AssetData({
+            active: true
+        });
+
+        emit AssetAdded(address(asset));
+    }
+
+    /**
+     * @notice Disables an existing asset in the system.
+     * @dev Sets an asset as inactive in the _assetData mapping. This function can only be called by the strategy manager.
+     * @param asset The address of the ERC20 token to be disabled.
+     */
+    function disableAsset(IERC20 asset) public onlyRole(ASSET_MANAGER_ROLE) whenNotPaused {
+        require(_assetData[asset].active, "Asset already inactive");
+
+        _assetData[asset].active = false;
+
+        emit AssetDeactivated(address(asset));
+    }
+
+    /**
+     * @notice Deletes an asset from the system entirely.
+     * @dev Removes an asset from the _assetData mapping and the assets array. This function can only be called by the strategy manager.
+     * @param asset The address of the ERC20 token to be deleted.
+     */
+    function deleteAsset(IERC20 asset) public onlyRole(ASSET_MANAGER_ROLE) whenNotPaused {
+        if (!_assetData[asset].active) {
+            revert AssetNotActiveOrNonexistent(address(asset));
+        }
```

Comment:
minor cosmetic note... A custom error is used here, but simular calls above use require statements. I'm not sure how important consistency is for these admin calls, or which style to use when and why.

Reply:
We should default to named errors you're right